--- pages/api/auth/[...nextauth].ts
+++ pages/api/auth/[...nextauth].ts
@@ -209,8 +209,8 @@
       options: {
         httpOnly: true,
         sameSite: "lax",
         path: "/",
-        domain: VERCEL_DEPLOYMENT ? ".papermark.com" : undefined,
-        secure: VERCEL_DEPLOYMENT,
+        domain: process.env.NEXTAUTH_COOKIE_DOMAIN || (VERCEL_DEPLOYMENT ? ".papermark.com" : undefined),
+        secure: process.env.NEXTAUTH_URL?.startsWith("https://") || VERCEL_DEPLOYMENT,
       },
     },
   },
