--- lib/resend.ts.orig
+++ lib/resend.ts
@@ -48,11 +48,11 @@
   const fromAddress =
     from ??
     (marketing
-      ? "Marc from Papermark <marc@updates.papermark.com>"
+      ? process.env.EMAIL_FROM || "Marc from Papermark <marc@updates.papermark.com>"
       : system
-        ? "Papermark <system@papermark.com>"
+        ? process.env.EMAIL_FROM || "Papermark <system@papermark.com>"
         : verify
-          ? "Papermark <system@verify.papermark.com>"
+          ? process.env.EMAIL_FROM || "Papermark <system@verify.papermark.com>"
           : !! scheduledAt
             ? "Marc Seitz <marc@papermark.com>"
             : "Marc from Papermark <marc@papermark.io>");
